Error: The terraform-provider-aws_v6.22.0_x5 plugin crashed!

```
31mâ•·[0m[0m
[31mâ”‚[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
[31mâ”‚[0m [0m
[31mâ”‚[0m [0m[0m  with aws_eks_pod_identity_association.vpc_cni_pod_identity,
[31mâ”‚[0m [0m  on main.tf line 638, in resource "aws_eks_pod_identity_association" "vpc_cni_pod_identity":
[31mâ”‚[0m [0m 638: resource "aws_eks_pod_identity_association" "vpc_cni_pod_identity" [4m{[0m[0m
[31mâ”‚[0m [0m
[31mâ”‚[0m [0mThe plugin encountered an error, and failed to respond to the
[31mâ”‚[0m [0mplugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain
[31mâ”‚[0m [0mmore details.
[31mâ•µ[0m[0m
[31mâ•·[0m[0m
[31mâ”‚[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
[31mâ”‚[0m [0m
[31mâ”‚[0m [0m[0m  with aws_eks_pod_identity_association.ebs_csi_pod_identity,
[31mâ”‚[0m [0m  on main.tf line 651, in resource "aws_eks_pod_identity_association" "ebs_csi_pod_identity":
[31mâ”‚[0m [0m 651: resource "aws_eks_pod_identity_association" "ebs_csi_pod_identity" [4m{[0m[0m
[31mâ”‚[0m [0m
[31mâ”‚[0m [0mThe plugin encountered an error, and failed to respond to the
[31mâ”‚[0m [0mplugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain
[31mâ”‚[0m [0mmore details.
[31mâ•µ[0m[0m
[31mâ•·[0m[0m
[31mâ”‚[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
[31mâ”‚[0m [0m
[31mâ”‚[0m [0m[0m  with aws_eks_pod_identity_association.external_dns_pod_identity,
[31mâ”‚[0m [0m  on main.tf line 664, in resource "aws_eks_pod_identity_association" "external_dns_pod_identity":
[31mâ”‚[0m [0m 664: resource "aws_eks_pod_identity_association" "external_dns_pod_identity" [4m{[0m[0m
[31mâ”‚[0m [0m
[31mâ”‚[0m [0mThe plugin encountered an error, and failed to respond to the
[31mâ”‚[0m [0mplugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain
[31mâ”‚[0m [0mmore details.
[31mâ•µ[0m[0m

Stack trace from the terraform-provider-aws_v6.22.0_x5 plugin:

panic: Value Conversion Error
	
	An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to the provider developer:
	
	Received unknown value, however the target type cannot handle unknown values. Use the corresponding `types` package type or a custom type that handles unknown values.
	
	Path: ["ManagementPublicIP"]
	Target Type: *string
	Suggested Type: basetypes.StringValue
	["ManagementPublicIP"]

goroutine 513 [running]:
github.com/hashicorp/terraform-provider-aws/internal/errs.Must[...](...)
	github.com/hashicorp/terraform-provider-aws/internal/errs/must.go:13
github.com/hashicorp/terraform-provider-aws/internal/errs/fwdiag.Must[...]({0x0?, 0x0}, {0xc0019d0510, 0x1b351940?, 0xc001f01d20?})
	github.com/hashicorp/terraform-provider-aws/internal/errs/fwdiag/must.go:17 +0x4b
github.com/hashicorp/terraform-provider-aws/internal/framework/flex.must(...)
	github.com/hashicorp/terraform-provider-aws/internal/framework/flex/errs.go:13
github.com/hashicorp/terraform-provider-aws/internal/framework/flex.ExpandFrameworkStringMap({0x21159fb8, 0xc0023ea6c0}, {0x211f3790, 0xc0018ed1c0})
	github.com/hashicorp/terraform-provider-aws/internal/framework/flex/map.go:17 +0x85
github.com/hashicorp/terraform-provider-aws/internal/tags.New({0x21159fb8, 0xc0023ea6c0}, {0x1e157540?, 0xc001c507b0})
	github.com/hashicorp/terraform-provider-aws/internal/tags/key_value_tags.go:650 +0x3c8
github.com/hashicorp/terraform-provider-aws/internal/provider/framework.resourceValidateRequiredTagsInterceptor.modifyPlan({}, {0x21159fb8, 0xc0023ea6c0}, {{0x21243ca0, 0xc0003ab2c0}, 0xc001c68380, 0xc0003f3080, 0x1})
	github.com/hashicorp/terraform-provider-aws/internal/provider/framework/tags_interceptor.go:295 +0x63b
github.com/hashicorp/terraform-provider-aws/internal/provider/framework.(*wrappedResource).ModifyPlan.(*wrappedResource).ModifyPlan.interceptedHandler[...].func2-range1(...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/framework/intercept.go:395
github.com/hashicorp/terraform-provider-aws/internal/provider/framework.(*wrappedResource).ModifyPlan.(*wrappedResource).ModifyPlan.interceptedHandler[...].func3(...)
	slices/iter.go:40
github.com/hashicorp/terraform-provider-aws/internal/provider/framework.(*wrappedResource).ModifyPlan.interceptedHandler[...].func2(...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/framework/intercept.go:394
github.com/hashicorp/terraform-provider-aws/internal/provider/framework.(*wrappedResource).ModifyPlan(0xc0024de000, {0x21159fb8?, 0xc0017e2b10?}, {{{{0x21237ec0, 0xc002aa3a10}, {0x1ce61820, 0xc002aa3740}}, {0x21282198, 0xc00502f8b0}}, {{{0x21237ec0, ...}, ...}, ...}, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/framework/wrap.go:733 +0x37d
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).PlanResourceChange(0xc00287fb08, {0x21159fb8, 0xc0017e2b10}, 0xc000c1e2a0, 0xc0022414e0)
	github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/fwserver/server_planresourcechange.go:347 +0x1f9b
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).PlanResourceChange(0xc00287fb08, {0x21159fb8?, 0xc0017e24e0?}, 0xc004ef3560)
	github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/proto5server/server_planresourcechange.go:72 +0x592
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).PlanResourceChange(0xc001e8f8c0, {0x21159fb8?, 0xc0017e20c0?}, 0xc004ef3560)
	github.com/hashicorp/terraform-plugin-mux@v0.21.0/tf5muxserver/mux_server_PlanResourceChange.go:73 +0x2a2
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0xc001e22c80, {0x21159fb8?, 0xc001199290?}, 0xc0023f6680)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/tf5server/server.go:904 +0x3d9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0x1e6b7220, 0xc001e22c80}, {0x21159fb8, 0xc001199290}, 0xc0023f6600, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:771 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00225da00, {0x21159fb8, 0xc001198ff0}, 0xc00508a960, 0xc003212ff0, 0x2f903cf0, 0x0)
	google.golang.org/grpc@v1.75.1/server.go:1431 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc00225da00, {0x2115aab8, 0xc002d384e0}, 0xc00508a960)
	google.golang.org/grpc@v1.75.1/server.go:1842 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.75.1/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 25
	google.golang.org/grpc@v1.75.1/server.go:1072 +0x11d

Error: The terraform-provider-aws_v6.22.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/.changelog/45201.txt
+++ b/.changelog/45201.txt
@@ -1,3 +1,6 @@
 ```release-note:bug
 provider: Fix early return logic in the required tag validation interceptor. This addresses a performance regression introduced in [v6.22.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#6220-november-20-2025).
 ```
+```release-note:bug
+provider: Fix crash in required tag validation interceptor when tag values are unknown. This addresses a regression introduced in [v6.22.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#6220-november-20-2025).
+```

--- a/internal/provider/framework/tags_interceptor.go
+++ b/internal/provider/framework/tags_interceptor.go
@@ -301,6 +301,10 @@ func (r resourceValidateRequiredTagsInterceptor) modifyPlan(ctx context.Context,
 			return
 		}
 
+		if !planTags.IsWhollyKnown() {
+			return
+		}
+
 		allPlanTags := c.DefaultTagsConfig(ctx).MergeTags(tftags.New(ctx, planTags))
 		allStateTags := c.DefaultTagsConfig(ctx).MergeTags(tftags.New(ctx, stateTags))
 

--- a/internal/provider/framework/tags_interceptor_test.go
+++ b/internal/provider/framework/tags_interceptor_test.go
@@ -91,11 +91,41 @@ func Test_resourceValidateRequiredTagsInterceptor(t *testing.T) {
 		},
 	}
 
+	// Null tags
 	attrs := map[string]tftypes.Value{
 		"name": tftypes.NewValue(tftypes.String, "test"),
 		"tags": tftypes.NewValue(tftypes.Map{ElementType: tftypes.String}, nil),
 	}
 	rawVal := tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), attrs)
+
+	// Partial required tags
+	attrsPartial := map[string]tftypes.Value{
+		"name": tftypes.NewValue(tftypes.String, "test"),
+		"tags": tftypes.NewValue(tftypes.Map{ElementType: tftypes.String}, map[string]tftypes.Value{
+			"bar": tftypes.NewValue(tftypes.String, nil),
+		}),
+	}
+	rawValPartial := tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), attrsPartial)
+
+	// All required tags
+	attrsRequired := map[string]tftypes.Value{
+		"name": tftypes.NewValue(tftypes.String, "test"),
+		"tags": tftypes.NewValue(tftypes.Map{ElementType: tftypes.String}, map[string]tftypes.Value{
+			"foo": tftypes.NewValue(tftypes.String, nil),
+			"bar": tftypes.NewValue(tftypes.String, nil),
+		}),
+	}
+	rawValRequired := tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), attrsRequired)
+
+	// Unknown tag values
+	attrsUnknown := map[string]tftypes.Value{
+		"name": tftypes.NewValue(tftypes.String, "test"),
+		"tags": tftypes.NewValue(tftypes.Map{ElementType: tftypes.String}, map[string]tftypes.Value{
+			"foo": tftypes.NewValue(tftypes.String, tftypes.UnknownValue),
+			"bar": tftypes.NewValue(tftypes.String, tftypes.UnknownValue),
+		}),
+	}
+	rawValUnknown := tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), attrsUnknown)
 
 	tests := []struct {
 		name      string
@@ -136,6 +166,93 @@ func Test_resourceValidateRequiredTagsInterceptor(t *testing.T) {
 			},
 		},
 		{
+			name: "create, partial tags",
+			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
+				c: mockRequiredTagsClient{},
+				request: &resource.ModifyPlanRequest{
+					Config: tfsdk.Config{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+					State: tfsdk.State{
+						Raw:    tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), nil), // Raw state is null on creation
+						Schema: resourceSchema,
+					},
+					Plan: tfsdk.Plan{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+				},
+				response: &resource.ModifyPlanResponse{
+					Plan: tfsdk.Plan{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+				},
+				when: Before,
+			},
+			wantDiags: diag.Diagnostics{diag.NewAttributeErrorDiagnostic(
+				path.Root(names.AttrTags),
+				"Missing Required Tags",
+				"An organizational tag policy requires the following tags for aws_test: [foo]",
+			),
+			},
+		},
+		{
+			name: "create, required tags",
+			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
+				c: mockRequiredTagsClient{},
+				request: &resource.ModifyPlanRequest{
+					Config: tfsdk.Config{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+					State: tfsdk.State{
+						Raw:    tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), nil), // Raw state is null on creation
+						Schema: resourceSchema,
+					},
+					Plan: tfsdk.Plan{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+				},
+				response: &resource.ModifyPlanResponse{
+					Plan: tfsdk.Plan{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+				},
+				when: Before,
+			},
+		},
+		{
+			name: "create, unknown tag values",
+			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
+				c: mockRequiredTagsClient{},
+				request: &resource.ModifyPlanRequest{
+					Config: tfsdk.Config{
+						Raw:    rawValUnknown,
+						Schema: resourceSchema,
+					},
+					State: tfsdk.State{
+						Raw:    tftypes.NewValue(resourceSchema.Type().TerraformType(ctx), nil), // Raw state is null on creation
+						Schema: resourceSchema,
+					},
+					Plan: tfsdk.Plan{
+						Raw:    rawValUnknown,
+						Schema: resourceSchema,
+					},
+				},
+				response: &resource.ModifyPlanResponse{
+					Plan: tfsdk.Plan{
+						Raw:    rawValUnknown,
+						Schema: resourceSchema,
+					},
+				},
+				when: Before,
+			},
+		},
+		{
 			name: "update, no tags change",
 			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
 				c: mockRequiredTagsClient{},
@@ -156,6 +273,93 @@ func Test_resourceValidateRequiredTagsInterceptor(t *testing.T) {
 				response: &resource.ModifyPlanResponse{
 					Plan: tfsdk.Plan{
 						Raw:    rawVal,
+						Schema: resourceSchema,
+					},
+				},
+				when: Before,
+			},
+		},
+		{
+			name: "update, add required",
+			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
+				c: mockRequiredTagsClient{},
+				request: &resource.ModifyPlanRequest{
+					Config: tfsdk.Config{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+					State: tfsdk.State{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+					Plan: tfsdk.Plan{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+				},
+				response: &resource.ModifyPlanResponse{
+					Plan: tfsdk.Plan{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+				},
+				when: Before,
+			},
+		},
+		{
+			name: "update, remove required",
+			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
+				c: mockRequiredTagsClient{},
+				request: &resource.ModifyPlanRequest{
+					Config: tfsdk.Config{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+					State: tfsdk.State{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+					Plan: tfsdk.Plan{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+				},
+				response: &resource.ModifyPlanResponse{
+					Plan: tfsdk.Plan{
+						Raw:    rawValRequired,
+						Schema: resourceSchema,
+					},
+				},
+				when: Before,
+			},
+			wantDiags: diag.Diagnostics{diag.NewAttributeErrorDiagnostic(
+				path.Root(names.AttrTags),
+				"Missing Required Tags",
+				"An organizational tag policy requires the following tags for aws_test: [foo]",
+			),
+			},
+		},
+		{
+			name: "update, unknown tag values",
+			opts: interceptorOptions[resource.ModifyPlanRequest, resource.ModifyPlanResponse]{
+				c: mockRequiredTagsClient{},
+				request: &resource.ModifyPlanRequest{
+					Config: tfsdk.Config{
+						Raw:    rawValUnknown,
+						Schema: resourceSchema,
+					},
+					State: tfsdk.State{
+						Raw:    rawValPartial,
+						Schema: resourceSchema,
+					},
+					Plan: tfsdk.Plan{
+						Raw:    rawValUnknown,
+						Schema: resourceSchema,
+					},
+				},
+				response: &resource.ModifyPlanResponse{
+					Plan: tfsdk.Plan{
+						Raw:    rawValUnknown,
 						Schema: resourceSchema,
 					},
 				},

--- a/internal/provider/sdkv2/tags_interceptor.go
+++ b/internal/provider/sdkv2/tags_interceptor.go
@@ -329,6 +329,10 @@ func validateRequiredTags() customizeDiffInterceptor {
 					return nil
 				}
 
+				if !d.GetRawPlan().GetAttr(names.AttrTags).IsWhollyKnown() {
+					return nil
+				}
+
 				cfgTags := tftags.New(ctx, d.Get(names.AttrTags).(map[string]any))
 				allTags := c.DefaultTagsConfig(ctx).MergeTags(cfgTags)
 				if allTags.ContainsAllKeys(reqTags) {

